### PR TITLE
Update register endpoint

### PR DESCRIPTION
**Pull request by KarolDenst**

### CODE REVIEW

File: backend/YouTubeV2.Application/Services/AzureServices/BlobServices/IBlobImageService.cs [hunk 1]
```diff
@@ -4,5 +4,6 @@ public interface IBlobImageService
     {
         Task UploadImageAsync(byte[] fileContent, string fileName, string blobContainerName, CancellationToken cancellationToken = default);
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/Services/AzureServices/BlobServices/IBlobImageService.cs]: probably we should  delete UploadImageAsync from interface and add UploadVideoThumbnailAsync. Also we should do similar thing with GetImageUrl

File: backend/YouTubeV2.Application/Services/UserService.cs [hunk 2]
```diff
@@ -27,13 +32,17 @@ public async Task RegisterAsync(RegisterDto registerDto, CancellationToken cance
             if (!result.Succeeded)
                 throw new BadRequestException(result.Errors.Select(error => new ErrorResponseDTO(error.Description)));
 
-            result = await _userManager.AddToRoleAsync(user, Role.User);
-            if (!result.Succeeded)
-                throw new BadRequestException(result.Errors.Select(error => new ErrorResponseDTO(error.Description)));
+            if (registerDto.userType.ToUpper() == Role.Simple.ToUpper()) result = await _userManager.AddToRoleAsync(user, Role.Simple);
+            else if (registerDto.userType.ToUpper() == Role.Creator.ToUpper()) result = await _userManager.AddToRoleAsync(user, Role.Creator);
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/Services/UserService.cs]: shouldn't creator have simple role too?

Comment (Marvvvin5) [reply backend/YouTubeV2.Application/Services/UserService.cs]: registerDto.userType.Equals(Role.Simple, StringComparison.IgnoreCase) or something like that don't remember the exact syntax

File: backend/YouTubeV2.Application/Validator/RegisterDtoValidator.cs [hunk 3]
```diff
@@ -12,6 +12,7 @@ public RegisterDtoValidator(UserManager<User> userManager)
         {
             RuleFor(x => x.name).NotNull().Length(1, UserConstants.MaxUserNameLength);
             RuleFor(x => x.surname).NotNull().Length(1, UserConstants.MaxUserSurnameLength);
+            RuleFor(x => x.userType).Must(userType => userType.ToUpper() == Role.Simple.ToUpper() || userType.ToUpper() == Role.Creator.ToUpper());
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/Validator/RegisterDtoValidator.cs]: same here

File: frontend/src/app/pages/registration/registration.component.html [hunk 4]
```diff
@@ -153,6 +153,30 @@
...
+      <div class="p-fluid">
+        <label for="avatarImage-input" class="field-label">
+          <p>Avatar Image</p>
+        </label>
+        <p-fileUpload
+          #fileUpload
+          name="signedContract[]"
+          accept="image/png, image/gif, image/jpeg"
+          [fileLimit]="1"
+          invalidFileTypeMessageSummary="Allowed file type is .png/.jpeg"
```
Comment (Marvvvin5) [inline frontend/src/app/pages/registration/registration.component.html:171]: what about gif? also idk how will blob storage work with other files than png. Do you know what file types are to be suported based on documentation?

File: frontend/src/app/pages/registration/registration.component.html [hunk 5]
```diff
@@ -153,6 +153,30 @@
...
+        <label for="userType-input" class="field-label">
+          <p>Creator</p>
+        </label>
+        <p-inputSwitch id="userType-input" formControlName="userType"/>
+      </div>
+      <div class="p-fluid">
+        <label for="avatarImage-input" class="field-label">
+          <p>Avatar Image</p>
+        </label>
+        <p-fileUpload
```
Comment (Marvvvin5) [inline frontend/src/app/pages/registration/registration.component.html:166]: this file upload looks bad ;(((((( You can have a look at the best application ever and check how it is getting done there

File: frontend/src/app/pages/registration/registration.component.ts [hunk 6]
```diff
@@ -46,7 +47,8 @@ export class RegistrationComponent implements OnInit, OnDestroy {
         Validators.pattern(this.containsNumberRegex),
         Validators.pattern(this.constainsSpecialCharacterRegex)
       ]),
-      confirmPassword: new FormControl('', [Validators.required])
+      confirmPassword: new FormControl('', [Validators.required]),
+      userType: new FormControl<boolean>(false),
```
Comment (Marvvvin5) [inline frontend/src/app/pages/registration/registration.component.ts:51]: what about image? what will happen when no image is provided and someone wants to access it later? Some default image? I guess the easiest solution is just to make it required

File: frontend/src/app/pages/registration/registration.component.html (hunk 4 above)
Comment (KarolDenst) [reply frontend/src/app/pages/registration/registration.component.html:171]: removed gif. It works with both png and jpeg

File: frontend/src/app/pages/registration/registration.component.ts (hunk 6 above)
Comment (KarolDenst) [reply frontend/src/app/pages/registration/registration.component.ts:51]: think its weird to force users to upload an image

Comment (KarolDenst) [reply frontend/src/app/pages/registration/registration.component.ts:51]: No way we have that in the specification

Comment (KarolDenst) [reply frontend/src/app/pages/registration/registration.component.ts:51]: btw it doesnt add image if we cont upload

**Review by Marvvvin5 — APPROVED** [review]

**Pull request merged**